fat/src/main.rs

```diff
@@ -79,16 +79,25 @@ async fn main() -> Result<()> {
 }
 
 async fn run(config: Config, db: DB, shutdown: Controller<String>) -> Result<()> {
+	// Fat client requires P2P to function properly, so we should exit if P2P is disabled
+	if matches!(config.network_mode, NetworkMode::RPCOnly) {
+		return Err(eyre!("Fat client cannot run in RPC-only mode. Please enable P2P by setting network_mode to 'Both' or 'P2POnly'."));
+	}
 	let version = clap::crate_version!();
 	let rev = env!("GIT_COMMIT_HASH");
 	info!(version, rev, "Running {}", clap::crate_name!());
 	info!("Using configuration: {config:?}");
 
-	let (p2p_keypair, p2p_peer_id) = p2p::identity(&config.libp2p, db.clone())?;
 	let partition = config.fat.block_matrix_partition;
 	let partition_size = format!("{}/{}", partition.number, partition.fraction);
 	let identity_cfg = IdentityConfig::from_suri("//Alice".to_string(), None)?;
 
+	// Initialize p2p components only if not in RPC-only mode
+	if matches!(config.network_mode, NetworkMode::RPCOnly) {
```

Comment:
Can we use equality here, since there is no need for a match (no complex enum variants)?